src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java

```diff
@@ -77,6 +119,16 @@ public synchronized Optional<Message> poll() {
         LOGGER.info("Message is: {}", polledMessage);
         LOGGER.info("Records for tablet: {}", mergeSlots.get(polledMessage.tablet).size());
         mergeSlots.get(polledMessage.tablet).removeIf(item -> item.compareTo(polledMessage) == 0);
+
+        // After removing the record, if there is any record left in the slot then update the tablet
+        // safetime to the value of the first record in the merge slot, iff the tablet safetime is
+        // less than that of the next message in the slot.
+        if (!mergeSlots.get(polledMessage.tablet).isEmpty()
```

Comment:
IIRC we had decided to remove this block? Is it still required?

Reply:
Removing this block was still causing the connector getting stuck issue, so this time I added the condition to only update on removal if the first message's commit time is higher than the tablet safetime so that we do not get stuck and that helped.